src/classes/VendorCommunicator.ts

```diff
@@ -39,47 +61,80 @@ interface TrackingEvent {
   actionData: Record<string, unknown>;
 }
 
+interface ProjectSavedPayload {
+  message: string;
+  customerId: string;
+  projectId: string;
+  versionId: string;
+}
+
+interface ProjectDeletedPayload {
+  message: string;
+  customerId: string;
+  projectId: string;
+}
+
+interface DirtyStateChangedPayload {
+  message: string;
+  isDirty: boolean;
+}
+
+interface iFrameRedirectPayload {
```

Comment:
I have a small preference to keep them separate, because we want to see a clear difference between an "application exit" event and a navigation tracking event. It's just a convenience on our side for developers to be able to see when an event is expected to trigger a redirect.